src/broadcast/scrobblingmanager.cpp

```diff
@@ -0,0 +1,258 @@
+#include "broadcast/scrobblingmanager.h"
+
+#include <QObject>
+
+#include "broadcast/filelistener/filelistener.h"
+#include "broadcast/listenbrainzlistener/listenbrainzservice.h"
+#ifdef __MPRIS__
+#include "broadcast/mpris/mprisservice.h"
+#endif
+#include "control/controlproxy.h"
+#include "engine/enginexfader.h"
+#include "mixer/deck.h"
+#include "mixer/playerinfo.h"
+#include "mixer/playermanager.h"
+#include "moc_scrobblingmanager.cpp"
+
+#ifdef MIXXX_BUILD_DEBUG
+QDebug operator<<(QDebug debug, const ScrobblingManager::TrackInfo& info) {
+    debug << "Pointer to track:" << info.m_trackInfo.get();
+    return debug << "Players:" << info.m_players;
+}
+#endif
+
+TotalVolumeThreshold::TotalVolumeThreshold(QObject* parent, double threshold)
+        : m_CPCrossfader("[Master]", "crossfader", parent),
+          m_CPXFaderCurve(ConfigKey(EngineXfader::kXfaderConfigKey,
+                                  "xFaderCurve"),
+                  parent),
+          m_CPXFaderCalibration(ConfigKey(EngineXfader::kXfaderConfigKey,
+                                        "xFaderCalibration"),
+                  parent),
+          m_CPXFaderMode(ConfigKey(EngineXfader::kXfaderConfigKey,
+                                 "xFaderMode"),
+                  parent),
+          m_CPXFaderReverse(ConfigKey(EngineXfader::kXfaderConfigKey,
+                                    "xFaderReverse"),
+                  parent),
+          m_pParent(parent),
+          m_volumeThreshold(threshold) {
+}
+
+bool TotalVolumeThreshold::isPlayerAudible(BaseTrackPlayer* pPlayer) const {
+    DEBUG_ASSERT(pPlayer);
+    double finalVolume;
+    ControlProxy trackPreGain(pPlayer->getGroup(), "pregain", m_pParent);
+    double preGain = trackPreGain.get();
+    ControlProxy trackVolume(pPlayer->getGroup(), "volume", m_pParent);
+    double volume = trackVolume.get();
+    ControlProxy deckOrientation(pPlayer->getGroup(), "orientation", m_pParent);
+    int orientation = static_cast<int>(deckOrientation.get());
+
+    CSAMPLE_GAIN xFaderLeft;
+    CSAMPLE_GAIN xFaderRight;
+
+    EngineXfader::getXfadeGains(m_CPCrossfader.get(),
+            m_CPXFaderCurve.get(),
+            m_CPXFaderCalibration.get(),
+            m_CPXFaderMode.get(),
+            m_CPXFaderReverse.toBool(),
+            &xFaderLeft,
+            &xFaderRight);
+    finalVolume = preGain * volume;
+    if (orientation == EngineChannel::LEFT) {
+        finalVolume *= xFaderLeft;
+    } else if (orientation == EngineChannel::RIGHT) {
+        finalVolume *= xFaderRight;
+    }
+    return finalVolume > m_volumeThreshold;
+}
+
+void TotalVolumeThreshold::setVolumeThreshold(double volume) {
+    m_volumeThreshold = volume;
+}
+
+ScrobblingManager::ScrobblingManager(UserSettingsPointer pConfig,
+        std::shared_ptr<PlayerManager> pPlayerManager,
+        MixxxMainWindow* pWindow)
```

Comment:
unused, please remove. Let's decouple application logic from QWidgets as much as possible.

Reply:
It is used for the window control in mpris.